packages/utils/upgrade/src/core/requirements/is-clean-git-repo.ts

```diff
@@ -0,0 +1,53 @@
+import assert from 'node:assert';
+import { simpleGit } from 'simple-git';
+import type { TaskOptions } from '../../types';
+
+type Params = Pick<TaskOptions, 'confirm' | 'logger'> & {
+  force: boolean;
+  cwd: string;
+};
+
+export const isCleanGitRepo = async ({ cwd, logger, force, confirm }: Params) => {
+  const git = simpleGit({ baseDir: cwd });
+  const repoStatus = {
+    isRepo: true,
+    isClean: true,
+    isGitInstalled: true,
+  };
+
+  try {
+    // Check if Git is installed
+    await git.version();
+    // Check if the path is under version control
+    repoStatus.isRepo = await git.checkIsRepo();
```

Comment:
Does this check if the current working directory contains a .git folder or does it detect .git folder higher in the file structure (e.g. mono-repositories?)

Reply:
It works with the mono repo too